src/app/lib/utilities/preprocessor/rules/blockPositioning.js

```diff
@@ -0,0 +1,49 @@
+import deepGet from '../../../../helpers/json/deepGet';
+import deepClone from '../../../../helpers/json/deepClone';
+
+const insertBlockPositioning = (
+  { model },
+  positionArr = [],
+) => {
+  if (!model || typeof model !== 'object') {
+    return null;
+  }
```

Comment:
```suggestion
  } else if (!('blocks' in model)) {
    return null;
  }
```
This allows us to remove the `forEach` on `Object.keys` by ensuring that `model.blocks` exists before we explicitly map on it. 

The checking on the `blocks` key on model needs to be in a different if statement to the `!model` check, otherwise there will be an `Uncaught TypeError: Cannot read property 'blocks' of undefined`. We could use something like [lodash `has()`](https://lodash.com/docs/4.17.11#has) to have a single statement which will gracefully check that model exists, is an object and has the key `blocks`. 
```
if (!has(model, 'blocks')) {
  return null;
}
```
I don't mind either way, but if you choose `has()` we should consider how heavy the function is. We could always copy and paste in the function into a helper and then has it re-usable for future cases.

Reply:
So looks like we already have a helper for this, similar to my suggestion, which is already being imported and used in this file (should probably of remembered that 🤦‍♂). `deepGet` ["Safely retrieves a nested object value, or `null` if doesn't exist."](https://github.com/bbc/simorgh/blob/1765_lazyloading/src/app/helpers/json/deepGet.js#L2) so this can be reduced to: 
```
if (!deepGet(['blocks'], model)) {
  return null;
}

model['blocks'].map((childObj, index) => { ...
```

Feel free to ignore all the suggestions and manually change the code 😎

Reply:
Much prefer this. Will address in the morning.

Reply:
WRT `deepGet(['blocks'], model)` - this method performs a `reduce`, which is a method on an array, not an object. In both instances that you've suggested using it (on `content`, and on `model`), these are objects, therefore we cannot use it here. I've opted for `'blocks' in model` and `'model' in content` however.

Reply:
From what I can tell the `reduce` is performed on the array of paths you pass to the function EG: `['key1', 'key2'], going to try it on the branch locally

Reply:
Ignore me. I wasn't passing an array.